src/components/Modal/modal.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { Keys } from '../../types';
+import { MODAL } from '../ThemeIdentifiers';
+import KeypressListener from '../KeypressListener';
+import Helpers from './helpers';
+import Dialog from './dialog';
+import * as baseTheme from './Modal.scss';
+
+const bodyStyle = (pading: string, overflow: string): void => {
+  const body = document.getElementsByTagName('body');
+  body[0].style.paddingRight = pading;
+  body[0].style.overflow = overflow;
+};
+
+const getModalElement = (id: string) => {
+  return {
+    modal: Helpers.getElement(`modal-${id}`),
+    dialog: Helpers.getElement(`dialog-${id}`),
+  };
+};
+
+export type Size = 'Small' | 'Medium' | 'Large' | number;
+
+export interface Ianimate {
+  in: Function,
+  out: Function,
+}
+
+export interface Itrigger {
+  body: string,
+  animate: Ianimate,
+}
+
+export interface Props {
+  className?: string,
+  classes?: string,
+  close?: boolean,
+  footer?: React.ReactNode,
+  header?: string | React.ReactNode,
+  activator: React.ReactElement<any>,
+  id?: string,
+  show: boolean,
+  trigger: Itrigger,
+  closeOnEsc?: boolean,
+  closeOnBackgroud?: boolean,
+  modalOverflow?: boolean,
+  backdropEnabled?: boolean,
+  size?: Size,
+  theme?: any,
+}
+
+class Modal extends React.Component<Props, {}> {
+  constructor(props: Props) {
+    super(props);
+
+    this.handleCloseClick = this.handleCloseClick.bind(this);
```

Comment:
Remove bind, and convert the functions below to arrow functions, which automatically bind.  Ex: `handleCloseClick = (e: any) => { ... }`